### PR TITLE
fix: upsert result use the previous pk

### PR DESCRIPTION
**Pull request by smellthemoon**
#34668

### CODE REVIEW

**Comment by smellthemoon** [comment]
rerun ut

File: internal/proxy/task_upsert.go [hunk 1]
```diff
@@ -63,6 +63,9 @@ type upsertTask struct {
 	schema           *schemaInfo
 	partitionKeyMode bool
 	partitionKeys    *schemapb.FieldData
+	// automatic generate pk as new pk wehen autoID == true
```
Comment (congqixia) [inline internal/proxy/task_upsert.go:66]: typo: `wehen` => `when`

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**